datafusion/expr/src/built_in_function.rs

```diff
@@ -1086,6 +1086,13 @@ impl BuiltinScalarFunction {
             BuiltinScalarFunction::RegexpReplace => Signature::one_of(
                 vec![
                     Exact(vec![Utf8, Utf8, Utf8]),
+                    Exact(vec![Binary, Utf8, Utf8]),
```

Comment:
I think it would be better to try and cast the binary --> utf8 at plan time rather than support a binary argument and then cast at plan time as it is more general and will apply to other functions

I left another comment below to this effect